Packages/MIES/MIES_Utilities.ipf

```diff
@@ -4562,23 +4574,76 @@ Function/WAVE FindLevelWrapper(data, level, edge)
 	ASSERT(numRows >= 2, "Expected wave with more than two rows")
 	ASSERT(IsFinite(level), "Expected finite level")
 	ASSERT(edge == FINDLEVEL_EDGE_INCREASING || edge == FINDLEVEL_EDGE_DECREASING || edge == FINDLEVEL_EDGE_BOTH, "Invalid edge type")
+	ASSERT(mode == FINDLEVEL_MODE_SINGLE || mode == FINDLEVEL_MODE_MULTI, "Invalid mode type")
 
 	ASSERT(DimSize(data, LAYERS) <= 1, "Unexpected input dimension")
+
 	Redimension/N=(numColsFixed * numRows)/E=1 data
```

Comment:
@t-b I Could we copy and work with the copy instead of modifying data passed into the function?